Convert rendered template to bytes before passing to napalm

barneysowood opened the pull request:
### What does this PR do?
When using python 3.x we need to pass a "bytes-like object" rather
than a string to load_merge_candidate or load_replace_candidate.

### What issues does this PR fix or reference?

Fixes #53115

### Previous Behavior
```root@34e639493580:/# salt \* state.apply test=True
[WARNING ] /usr/local/lib/python3.7/site-packages/salt/transport/ipc.py:292: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  self.unpacker = msgpack.Unpacker(encoding=encoding)

[WARNING ] /usr/local/lib/python3.7/site-packages/salt/payload.py:149: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  ret = msgpack.loads(msg, use_list=True, ext_hook=ext_type_decoder, encoding=encoding)

testrtr:
----------
          ID: manage_config
    Function: netconfig.managed
      Result: False
     Comment: Cannot execute "load_replace_candidate" on 192.168.122.251 as vyos. Reason: a bytes-like object is required, not 'str'!
              Configuration discarded.
     Started: 10:02:18.571889
    Duration: 13429.473 ms
     Changes:   

Summary for testrtr
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:  13.429 s
ERROR: Minions returned with non-zero exit code
```
### New Behavior
```
root@34e639493580:/# salt \* state.apply test=True
[WARNING ] /usr/local/lib/python3.7/site-packages/salt/transport/ipc.py:292: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  self.unpacker = msgpack.Unpacker(encoding=encoding)

[WARNING ] /usr/local/lib/python3.7/site-packages/salt/payload.py:149: DeprecationWarning: encoding is deprecated, Use raw=False instead.
  ret = msgpack.loads(msg, use_list=True, ext_hook=ext_type_decoder, encoding=encoding)

testrtr:
----------
          ID: manage_config
    Function: netconfig.managed
      Result: True
     Comment: Configuration discarded.
     Started: 12:48:19.981584
    Duration: 18490.453 ms
     Changes:   

Summary for testrtr
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:  18.490 s
```

### Tests written?

No

### Commits signed with GPG?

No

Ch3LL reviewed (changes requested):
Can you add a regression test to ensure this does not regress in the future?

Commit:
Convert rendered template to bytes before passing to napalm

When using python 3.x we need to pass a "bytes-like object" rather
than a string to load_merge_candidate or load_replace_candidate.
